tests/test_encoding.py

```diff
@@ -11,6 +11,23 @@
 from .test_helpers import ENCODING_FACTORIES, MAX_EXAMPLES
 
 
+@pytest.mark.skip(reason="Takes a really long time to finish, but was added to reproduce a crash.")
+@pytest.mark.parametrize("make_enc", ENCODING_FACTORIES)
+def test_extremely_big_encoding(make_enc: Callable[[], tiktoken.Encoding]):
+    enc = make_enc()
+    for c in ["^", "0", "a", "'s"]: # TODO " ", "\n" are still failing
```

Comment:
I have tried atomic grouping here (and a few other tricks), but couldn't make it work.
So all-whitespace humonguous inputs will still fail, everything else is likely fixed